sql/partitioning/partition_base.cc

```diff
@@ -392,12 +392,15 @@ Partition_base::Partition_base(handlerton *hton, TABLE_SHARE *share)
 }
 
 Partition_base::Partition_base(handlerton *hton, TABLE_SHARE *share,
-    Handler_share** ha_share)
+    Partition_base* clone_base,
+    MEM_ROOT* clone_mem_root)
     : handler(hton, share), Partition_helper(this)
 {
   DBUG_ENTER("Partition_base::Partition_base(table, ha_share)");
   init_handler_variables();
-  set_ha_share_ref(ha_share);
+  m_clone_base = clone_base;
```

Comment:
Use constructor initializer list

Reply:
well, init_handler_variables() would reset the value.

Ideally, init_handler_variables() should be removed and all variables should be moved to constructor initalizer list.